oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py

```diff
@@ -0,0 +1,589 @@
+import torch
+import itertools
+import torch.distributed as dist
+from functools import partial
+
+# from colossalai.zero.utils.zero_hook_v2 import ZeROHookV2
+# from colossalai.tensor.param_op_hook import ParamOpHookManager
+# from colossalai.gemini.gemini_mgr import GeminiManager
+from typing import Dict, Iterable, List, Optional, Set
+
+# from colossalai.logging import get_dist_logger
+from collections import OrderedDict
+
+# from colossalai.tensor.colo_parameter import ColoParameter, ColoTensor, ColoTensorSpec
+# from colossalai.tensor import ProcessGroup as ColoProcessGroup
+# from .reducer import Reducer
+from oslo.torch.nn.parallel.data_parallel._coloddp.reducer import Reducer
+
+# from colossalai.gemini.chunk import TensorState, Chunk, ChunkManager
+# from colossalai.nn.parallel.utils import get_temp_total_chunk_on_cuda
+from oslo.torch.distributed.parallel_context import ParallelContext
+from oslo.torch.distributed.parallel_mode import ParallelMode
+
+try:
+    from torch.nn.modules.module import _EXTRA_STATE_KEY_SUFFIX, _IncompatibleKeys
+except ImportError:
+    _EXTRA_STATE_KEY_SUFFIX = "_extra_state"
+
+
+def free_storage(data: torch.Tensor) -> None:
+    """Free underlying storage of a Tensor."""
+    if data.storage().size() > 0:
+        # Since we're modifying the Tensor's Storage directly, make sure the Tensor
+        # is the sole occupant of the Storage.
+        assert data.storage_offset() == 0
+        data.storage().resize_(0)
+
+
+def _cast_float(args, dtype: torch.dtype):
+    if isinstance(args, torch.Tensor) and torch.is_floating_point(args):
+        args = args.to(dtype)
+    elif isinstance(args, (list, tuple)):
+        args = type(args)(_cast_float(t, dtype) for t in args)
+    elif isinstance(args, dict):
+        args = {k: _cast_float(v, dtype) for k, v in args.items()}
+    return args
+
+
+class ColoDDP(torch.nn.Module):
```

Comment:
Let's use DistributedDataParallel as name.
remove all the `colo`.

Reply:
Done